### PR TITLE
check utf-8 encoding, fix for content-type

### PR DESCRIPTION
**Pull request by henderkes**
header content-type can now contain more than only 'application/json[5]', such as 'application/json; charset: utf-8' see #133

### CODE REVIEW

**Comment by Drakulix** [comment]
Maybe reword the error message to 'Only utf-8 compatible charsets are supported, the request body does not appear to be utf-8.'. The message might otherwise be quite confusing, if you e.g. specified charset=ascii in the content-type and the returned error is 'no valid utf-8'.

**Comment by henderkes** [comment]
Ugh, this took longer than expected but it should be ready and working now

**Review by Drakulix — APPROVED** [review]
